### PR TITLE
EWPP-1714: Generic webtools service.

### PR DESCRIPTION
**Pull request by yenyasinn**

### CODE REVIEW

**Review by upchuk — CHANGES_REQUESTED** [review]

File: modules/oe_webtools_media/tests/src/FunctionalJavascript/MediaSourceWebtoolsTest.php [hunk 1]
```diff
@@ -75,7 +76,7 @@ public function testMediaWebtoolsSource(): void {
       // Create a webtools media item with invalid webtools snippet.
       $this->drupalGet("media/add/{$media_type_id}");
       $assert_session->fieldExists('Name')->setValue("Invalid webtools $widget_name item");
-      $assert_session->fieldExists("Webtools {$widget_name} snippet")->setValue('{"service": "' . $service . $widget_type . '"}');
+      $assert_session->fieldExists("Webtools {$widget_name} snippet")->setValue('{"service": "' . $invalid_service . '"}');
```
Comment (22Alexandra) [inline modules/oe_webtools_media/tests/src/FunctionalJavascript/MediaSourceWebtoolsTest.php:79]: After adapting the validation messages we should cover both cases of invalid or blacklisted services.

**Review by 22Alexandra — CHANGES_REQUESTED** [review]

**Review by 22Alexandra — CHANGES_REQUESTED** [review]
Last changes are not good. We don't need 2 different constraints and all this refactoring. Add 2 different messages in the same constraint and for the validation display the proper message according to the conditions.

**Review by 22Alexandra — CHANGES_REQUESTED** [review]

File: composer.json [hunk 2]
```diff
@@ -76,7 +77,9 @@
             "Explicit requirement of drupal/token module due to PHP8 compatbility.",
             "Explicit requirement of behat module due to PHP8 compatibility.",
             "Explicit lower version requirement of drupal/ctools due to Drupal core 9.2 compatibility.",
-            "Explicit requirement for egulias/email-validator due to https://www.drupal.org/project/drupal/issues/3061074#comment-14300579. It can be removed when Drupal core 9.2 support is droppped."
+            "Explicit requirement for egulias/email-validator due to https://www.drupal.org/project/drupal/issues/3061074#comment-14300579. It can be removed when Drupal core 9.2 support is droppped.",
+            "Explicit requirement of phpmd/phpmd due to latest changes in composer",
```
Comment (22Alexandra) [inline composer.json:81]: Please remove this before merging.

**Review by 22Alexandra — APPROVED** [review]

File: composer.json [hunk 3]
```diff
@@ -13,6 +13,7 @@
     "require-dev": {
         "behat/behat": "^3.10",
         "composer/installers": "~1.5",
+        "composer/xdebug-handler": "^2.0",
```
Comment (22Alexandra) [inline composer.json:16]: Please remove this as it has been fixed in code-review: https://github.com/openeuropa/code-review/blob/2.x/composer.json#L7

**Review by 22Alexandra — DISMISSED** [review]

**Review by imanoleguskiza — APPROVED** [review]

**Review by imanoleguskiza — APPROVED** [review]

**Review by 22Alexandra — APPROVED** [review]

**Pull request merged**